cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/connector/ConnectorSink.java

```diff
@@ -66,18 +66,23 @@ public void prepareRun(BatchSinkContext context) throws Exception {
   }
 
   @Override
-  public void transform(StructuredRecord input, Emitter<KeyValue<NullWritable, Text>> emitter) throws Exception {
+  public void transform(KeyValue<String, StructuredRecord> input, Emitter<KeyValue<NullWritable, Text>> emitter)
+    throws Exception {
+    StructuredRecord taggedInput = input.getValue();
     if (writeSchema) {
-      input = modifyRecord(input);
+      taggedInput = modifyRecord(input);
     }
-    emitter.emit(new KeyValue<>(NullWritable.get(), new Text(StructuredRecordStringConverter.toJsonString(input))));
+    emitter.emit(new KeyValue<>(NullWritable.get(), new Text(StructuredRecordStringConverter.
+      toJsonString(taggedInput))));
```

Comment:
Since the name of the `taggedInput` is bigger than `input`, line goes beyond limit.